src/lib/components/custom/tab-bar/tab-bar.svelte

```diff
@@ -0,0 +1,40 @@
+<script lang="ts">
+	import { setContext } from 'svelte'
+	import { withTabStore, type TabStore } from './tab-store.svelte'
+	import { Button } from '$lib/components'
+
+	const store = withTabStore()
```

Comment:
I considered this, but right now it would make the code significantly more complex because currently the store items are populated as the child components are added, so we would need to track the initial state, the default value, the selected value depending if the user has already selected.

At the component page we don't need this feature so at the moment I don't think this is a good time investment. I would reconsider this when we want to make this a published component.